pkl/version.go

```diff
@@ -2,21 +2,67 @@ package pkl
 
 import (
 	"fmt"
+	"regexp"
 	"strconv"
+	"strings"
 )
 
+var semverPattern = regexp.MustCompile(`(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?`)
+
+var numericIdentifer = regexp.MustCompile(`^(0|[1-9]\d*)$`)
+
 //goland:noinspection GoSnakeCaseUsage
 var pklVersion0_25 = mustParseSemver("0.25.0")
 
 //goland:noinspection GoSnakeCaseUsage
 var pklVersion0_26 = mustParseSemver("0.26.0")
 
 type semver struct {
-	major      int
-	minor      int
-	patch      int
-	prerelease string
-	build      string
+	major                 int
+	minor                 int
+	patch                 int
+	prerelease            string
+	build                 string
+	prereleaseIdentifiers []prereleaseIdentifier
+}
+
+type prereleaseIdentifier struct {
+	numbericId     int
+	alphaNumericId string
+}
+
+func (i prereleaseIdentifier) compareTo(other prereleaseIdentifier) int {
+	if i.alphaNumericId != "" {
+		if other.alphaNumericId != "" {
+			return strings.Compare(i.alphaNumericId, other.alphaNumericId)
+		} else {
+			return 1
+		}
```

Comment:
This _is_ the behaviour of `strings.Compare`, no?
```suggestion
		return strings.Compare(i.alphaNumericId, other.alphaNumericId)
```

Reply:
Ah, great catch